### PR TITLE
Associations: has, belongs_to and many-to-many relations

### PR DESCRIPTION
**Pull request by AliRezaBmeDu**
In this branch I have done the following changes.

- Create a class Classroom with the following:

    1.     @label instance variable, should be initialized in the constructor.
    2.     Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

      1.     Create the has-many side (a classroom has many students).
      2.     Create the belongs-to side (a student belongs to a classroom).
      3.     Make sure that when adding a student to a classroom it also sets the classroom for the student.
      4.     Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Create a class Book with the following:

      1.     @title and @author instance variables, should be initialized in the constructor.
      2.     Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:

        1.     @date instance variable, should be initialized in the constructor.
        2.     Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

          1.     Create the has-many side of Book and Rental (a book has many rentals).
          2.     Create the belongs-to side of Rental and Book (a rental belongs to a book).
          3.     Create the has-many side of Person and Rental (a person has many rentals).
          4.     Create the belongs-to side of Rental and Person (a rental belongs to a person).
          5.     Modify the constructor of Rental so Book and Person are set in it.

- Add test files

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  attr_accessor :label, :student_group
+
+  def initialize(label)
+    @label = label
+    @student_group = [] # this will contain the students
+  end
+
+  def add_student(student)
+    @student_group << student # This will add a student objecgt to the students
+    student.classroom = label
```
Comment (DammyShittu) [inline classroom.rb]: - [x] Instead `label`, kindly assign the student's classroom to be the classroom itself.

  ```
  student.classroom = self
  ```

File: student.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 require_relative 'person'
 
 class Student < Person
```
Comment (DammyShittu) [inline student.rb:3]: - [x] The Student should have a method for the association.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -20,6 +21,10 @@ def correct_name
     name
   end
 
+  def add_rental(rental)
+    rentals << rental
```
Comment (DammyShittu) [inline person.rb]: - [x] The `rentals` here should be the instance variable.

  ```
  @rentals << rental
  ```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals << rental
```
Comment (DammyShittu) [inline book.rb]: - [x] The `rentals` here should be the instance variable.

  ```
  @rentals << rental
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @AliRezaBmeDu,

## Highlights
✔️ Association feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (AliRezaBmeDu) [reply classroom.rb]: Done

File: student.rb (hunk 2 above)
Comment (AliRezaBmeDu) [reply student.rb:3]: done

File: person.rb (hunk 3 above)
Comment (AliRezaBmeDu) [reply person.rb]: done

File: book.rb (hunk 4 above)
Comment (AliRezaBmeDu) [reply book.rb]: done

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED
Hi @AliRezaBmeDu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights

✔️ Linters passing
✔️ Correct GitFlow
✔️ Good commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**